povm_toolbox/quantum_info/multi_qubit_frame.py

```diff
@@ -107,11 +133,57 @@ def num_operators(self) -> int:
         """The number of effects of the frame."""
         return self._num_operators
 
+    @property
+    def shape(self) -> tuple[int, ...]:
+        """Return the shape of the frame."""
+        return self._shape or (self.num_operators,)
+
+    @shape.setter
+    def shape(self, new_shape: tuple[int, ...] | None) -> None:
+        """Set the shape of the frame."""
+        if new_shape is not None and prod(new_shape) != self.num_operators:
+            raise ValueError(
+                f"The shape {new_shape} is not compatible with the number of operators in the frame"
+                f" ({self.num_operators})."
+            )
+        self._shape = new_shape
+
     @property
     def operators(self) -> list[Operator]:
         """Return the list of frame operators."""
         return self._operators
 
+    @operators.setter
+    def operators(self, new_operators: list[Operator]):
+        """Set the frame operators."""
+        if self._shape is not None and prod(self._shape) != len(new_operators):
+            raise ValueError(
+                f"The number of operators ({len(new_operators)}) is not compatible with the shape"
+                f" of the frame: {self.shape}."
+            )
+        self._num_operators = len(new_operators)
+        self._dimension = new_operators[0].dim[0]
+        for frame_op in new_operators:
+            if not (self._dimension == frame_op.dim[0] and self._dimension == frame_op.dim[1]):
+                raise ValueError(
+                    f"Frame operators need to be square ({frame_op.dim[0]},{frame_op.dim[1]}) and "
+                    "all of the same dimension."
```

Comment:
Can you include the determined dimension here? Maybe like this?
```suggestion
                    f"all of the same dimension: {self._dimension}."
```